solr/core/src/java/org/apache/solr/handler/component/ResponseLogComponent.java

```diff
@@ -106,7 +107,7 @@ protected void processScores(
     StringBuilder sb = new StringBuilder();
     Set<String> fields = Collections.singleton(schema.getUniqueKeyField().getName());
     for (DocIterator iter = dl.iterator(); iter.hasNext(); ) {
-      sb.append(schema.printableUniqueKey(searcher.doc(iter.nextDoc(), fields)))
+      sb.append(schema.printableUniqueKey(searcher.storedFields().document(iter.nextDoc(), fields)))
```

Comment:
I think `searcher.storedFields()` can be pulled out of the for loop? probably reused with the one from line 95?